packages/synapse-react-client/src/synapse-queries/dataaccess/useAccessApprovals.ts

```diff
@@ -1,34 +1,48 @@
-import { useInfiniteQuery, UseInfiniteQueryOptions } from 'react-query'
+import {
+  InfiniteData,
+  QueryKey,
+  useInfiniteQuery,
+  UseInfiniteQueryOptions,
+} from '@tanstack/react-query'
 import SynapseClient from '../../synapse-client'
-import { SynapseClientError } from '../../utils/SynapseClientError'
-import { useSynapseContext } from '../../utils/context/SynapseContext'
+import { SynapseClientError, useSynapseContext } from '../../utils'
 import {
   AccessApprovalSearchRequest,
   AccessApprovalSearchResponse,
 } from '@sage-bionetworks/synapse-types'
 
-export function useSearchAccessApprovalsInfinite(
+export function useSearchAccessApprovalsInfinite<
+  TData = InfiniteData<AccessApprovalSearchResponse>,
+>(
   params: AccessApprovalSearchRequest,
   options?: UseInfiniteQueryOptions<
     AccessApprovalSearchResponse,
-    SynapseClientError
+    SynapseClientError,
+    TData,
+    AccessApprovalSearchResponse,
+    QueryKey,
+    AccessApprovalSearchResponse['nextPageToken']
```

Comment:
Our usage of `useInfiniteQuery` requires us to specify a type for `nextPageToken`, Unfortunately, it's the 6th generic type parameter in `UseInfiniteQueryOptions`, so you'll see a lot of this boilerplate for our custom `useInfiniteQuery` hooks.